script.js

```diff
@@ -1,4 +1,90 @@
+//Initiate variables and default game mode
+var gameMode = "P1RollDice";
+var P1Rolls = [];
+var P2Rolls = [];
+var P1Choice = 0;
+var P2Choice = 0;
+var numberOfDice = 2;
+
+//Function to roll x number of dice rolls
+var rollTwoDice = function (playerID) {
+  currentNumberOfDiceRolls = 0;
+  while (currentNumberOfDiceRolls < numberOfDice) {
+    var currentDiceRollNumber = Math.floor(Math.random() * 6) + 1;
+    playerID[currentNumberOfDiceRolls] = currentDiceRollNumber;
+    currentNumberOfDiceRolls = currentNumberOfDiceRolls + 1;
+  }
+};
+
 var main = function (input) {
-  var myOutputValue = 'hello world';
+  var myOutputValue = "";
+  //P1 Roll Dice
+  if (gameMode == "P1RollDice") {
+    console.log("Current Game Mode:" + gameMode);
+    rollTwoDice(P1Rolls);
+    console.log("P1 Dice Rolls:" + P1Rolls);
+    myOutputValue = "You rolled a " + P1Rolls[0] + " for Dice 1 and a " + P1Rolls[1] + " for Dice 2.<br>Choose the order of the dice by selecting whether 1 (Dice 1) or 2 (Dice 2) should be the first digit.";
+    gameMode = "P1ChooseOrder";
+    console.log("Current Game Mode:" + gameMode);
+    return myOutputValue;
+  }
+  //P1 Choose Order of Dice with input validation
+  if (gameMode == "P1ChooseOrder") {
+    if (input == 1) {
+      console.log("P1 Choose Which Dice First:" + input);
+      myOutputValue = "Your number is ".concat(P1Rolls[0], P1Rolls[1]) + "<br> It is now Player 2's turn. ";
+      P1Choice = Number("".concat(P1Rolls[0], P1Rolls[1]));
+      gameMode = "P2RollDice";
+      console.log("P1Choice:" + P1Choice);
+    } else if (input == 2) {
+      console.log("P1 Choose Which Dice First:" + input);
+      myOutputValue = "Your number is ".concat(P1Rolls[1], P1Rolls[0]) + "<br> It is now Player 2's turn. ";
+      P1Choice = Number("".concat(P1Rolls[1], P1Rolls[0]));
+      gameMode = "P2RollDice";
+      console.log("P1Choice:" + P1Choice);
+    } else {
+      myOutputValue = input + " is an invalid selection. <br> Please choose the order of the dice by selecting whether 1 (Dice 1) or 2 (Dice 2) should be the first digit.";
+    }
+    return myOutputValue;
+  }
+  //P2 Roll Dice
+  if (gameMode == "P2RollDice") {
+    console.log("Current Game Mode:" + gameMode);
+    rollTwoDice(P2Rolls);
+    console.log("P2 Dice Rolls:" + P2Rolls);
+    myOutputValue = "You rolled a " + P2Rolls[0] + " for Dice 1 and a " + P2Rolls[1] + " for Dice 2.<br>Choose the order of the dice by selecting whether 1 (Dice 1) or 2 (Dice 2) should be the first digit.";
+    gameMode = "P2ChooseOrder";
+    console.log("Current Game Mode:" + gameMode);
+    return myOutputValue;
```

Comment:
Regarding your question about the optimal position of `return`s, I think the way you have set it up is alright. 

This is because you want to change the game mode like you see on [line 56](https://github.com/rocketacademy/basics-beat-that/blob/644677a86aec7d0e0b5470eccc2426a27879392c/script.js#L56), without going into the next `if` block (where we check if `gameMode` is `P2ChooseOrder`) right away.

With this setup, unfortunately we will need to `return` within the `if` blocks to terminate the function early and show the copywriting (`myOutputValue`) relevant to the current game state. 

Without the `return` it will proceed on to the next game state (`"P2ChooseOrder"`) immediately , and the player will never getting to see the copywriting for the `P2Rolls`: `"You rolled a " + P2Rolls[0] + " for Dice 1 and a " + P2Rolls[1] + " for Dice 2. ..."`

If you are worried about the tidiness of your code, I will instead recommend you modularise each game state into a function. Let's take this `P2RollDice` game state for instance. Won't your code be neater if you can do it this way:

```
if (gameMode == "P2RollDice") {
  return p2RollDice();
}
```
where the logic in the `p2RollDice` function looks like this:

```
var p2RollDice = function () {
  // P2Rolls is a global variable, no need to be passed as param
  rollTwoDice(P2Rolls);
  // P2Rolls is a global variable, no need to be passed as param
  var output = "You rolled a " + P2Rolls[0] + " for Dice 1 and a " + P2Rolls[1] + " for Dice 2.<br>Choose the order of the dice by selecting whether 1 (Dice 1) or 2 (Dice 2) should be the first digit.";
  // gameMode is a global variable, no need to be passed as param
  gameMode = "P2ChooseOrder";
  return output
};
```